test/mapmatch.cc

```diff
@@ -1045,6 +1045,28 @@ TEST(Mapmatch, test_intersection_matching) {
   }
 }
 
+TEST(Mapmatch, test_degenerate_match) {
+  std::vector<std::string> test_cases = {
+      R"({"costing":"auto","format":"osrm","shape_match":"map_snap","shape":[
+          {"lat": 52.0981280, "lon": 5.1297250, "type": "break", "time":10},
+          {"lat": 52.0981280, "lon": 5.1297250, "type": "break", "time":169}],
+          "trace_options": {"interpolation_distance": 0}})",
```

Comment:
this is important, we never saw much of this before because people generally dont want to skip interpolation. i would argue they still dont :smile: